ibc-core/ics04-channel/types/src/error.rs

```diff
@@ -3,173 +3,132 @@
-    MissingHeight,
-    /// packet data bytes must be valid UTF-8 (this restriction will be lifted in the future)
-    NonUtf8PacketData,
+    /// invalid counterparty: expected `{expected}`, actual `{actual}`
+    InvalidCounterparty {
+        expected: Counterparty,
+        actual: Counterparty,
+    },
+    /// missing proof
+    MissingProof,
+    /// missing proof height
+    MissingProofHeight,
     /// missing counterparty
     MissingCounterparty,
+    /// missing channel end
+    MissingChannelEnd,
+    /// missing channel counter
+    MissingCounter,
     /// unsupported channel upgrade sequence
     UnsupportedChannelUpgradeSequence,
-    /// version not supported: expected `{expected}`, actual `{actual}`
-    VersionNotSupported { expected: Version, actual: Version },
-    /// missing channel end
-    MissingChannel,
-    /// the channel end (`{port_id}`, `{channel_id}`) does not exist
-    ChannelNotFound {
+    /// unsupported version: expected `{expected}`, actual `{actual}`
+    UnsupportedVersion { expected: Version, actual: Version },
+    /// non-existent channel end: (`{port_id}`, `{channel_id}`)
+    NonexistentChannel {
         port_id: PortId,
         channel_id: ChannelId,
     },
-    /// Verification fails for the packet with the sequence number `{sequence}`, error: `{client_error}`
-    PacketVerificationFailed {
+    /// packet data bytes must be valid UTF-8
+    NonUtf8PacketData,
+    /// failed packet verification for packet with sequence `{sequence}`: `{client_error}`
+    FailedPacketVerification {
         sequence: Sequence,
         client_error: ClientError,
     },
-    /// Error verifying channel state error: `{0}`
-    VerifyChannelFailed(ClientError),
-    /// String `{value}` cannot be converted to packet sequence, error: `{error}`
-    InvalidStringAsSequence {
-        value: String,
+    /// failed channel verification: `{0}`
+    FailedChannelVerification(ClientError),
+    /// failed to parse `{actual}` as packet sequence: `{error}`
+    FailedToParseSequence {
+        actual: String,
         error: core::num::ParseIntError,
     },
-    /// invalid channel counterparty: expected `{expected}`, actual `{actual}`
-    InvalidCounterparty {
-        expected: Counterparty,
-        actual: Counterparty,
-    },
-    /// application module error: `{description}`
-    AppModule { description: String },
-    /// Undefined counterparty connection for `{connection_id}`
-    UndefinedConnectionCounterparty { connection_id: ConnectionId },
-    /// invalid proof: empty proof
-    InvalidProof,
-    /// identifier error: `{0}`
-    InvalidIdentifier(IdentifierError),
-    /// channel counter overflow error
-    CounterOverflow,
-    /// other error: `{description}`
-    Other { description: String },
+    /// failed to update counter: `{description}`
+    FailedToUpdateCounter { description: String },
+    /// failed to store channel: `{description}`
+    FailedToStoreChannel { description: String },
```

Comment:
These are more like host-relevant errors. They should be removed from here once we differentiated between host and protocol errors.